### ISSUE TITLE
default configurePhase fails if there is no configure script

### ISSUE THREAD

**Issue by ericbmerritt** [NixOS/nixpkgs#12632]
This is in current master.

The problem is that if there is no configure script and `$dontDisableStatic` is not set, or is set to false. Then the configurePhase hangs. This is because the `grep -q enable-static $configureScript;` becomes `grep -q enable-static` if `$configureScript` is not set. Thats just grep waiting on input that never arrives.

The generated configure phase looks as follows:
```
onfigurePhase is a function
configurePhase () 
{ 
    runHook preConfigure;
    if [ -z "$configureScript" -a -x ./configure ]; then
        configureScript=./configure;
    fi;
    if [ -z "$dontFixLibtool" ]; then
        find . -iname "ltmain.sh" | while read i; do
            echo "fixing libtool script $i";
            fixLibtool $i;
        done;
    fi;
    if [ -z "$dontAddPrefix" ]; then
        configureFlags="${prefixKey:---prefix=}$prefix $configureFlags";
    fi;
    if [ -z "$dontAddDisableDepTrack" ]; then
        if grep -q dependency-tracking $configureScript; then
            configureFlags="--disable-dependency-tracking $configureFlags";
        fi;
    fi;
    if [ -z "$dontDisableStatic" ]; then
        if grep -q enable-static $configureScript; then
            configureFlags="--disable-static $configureFlags";
        fi;
    fi;
    if [ -n "$configureScript" ]; then
        echo "configure flags: $configureFlags ${configureFlagsArray[@]}";
        $configureScript $configureFlags "${configureFlagsArray[@]}";
    else
        echo "no configure script, doing nothing";
    fi;
    runHook postConfigure
}

``

**Comment by vcunat** [comment]
I  see, this is my fault, introduced recently. I should get to fix it properly within a few days, unless someone beats me to it.

**Comment by vcunat** [comment]
I would fix it like this:
```diff
diff --git a/pkgs/stdenv/generic/setup.sh b/pkgs/stdenv/generic/setup.sh
index 102a8f2..c4f9a6a 100644
--- a/pkgs/stdenv/generic/setup.sh
+++ b/pkgs/stdenv/generic/setup.sh
@@ -639,14 +639,14 @@ configurePhase() {
 
     # Add --disable-dependency-tracking to speed up some builds.
     if [ -z "$dontAddDisableDepTrack" ]; then
-        if grep -q dependency-tracking $configureScript; then
+        if grep -q dependency-tracking "$configureScript"; then
             configureFlags="--disable-dependency-tracking $configureFlags"
         fi
     fi
 
     # By default, disable static builds.
     if [ -z "$dontDisableStatic" ]; then
-        if grep -q enable-static $configureScript; then
+        if grep -q enable-static "$configureScript"; then
             configureFlags="--disable-static $configureFlags"
         fi
     fi
```

Would you also test it on your use case?

I think it's better to quote this variable in general, because it is common and even documented to pass space-separated commands in there. The greps should just fail in that case and `if` won't proceed which seems fine for such cases, and it's certainly better than passing additional unintended parameters to grep (which was happening all the time before).

**Comment by vcunat** [comment]
Fix in staging as 89036ef76a. Reopen if you think this wasn't fixed.
